### ISSUE TITLE
[Accessibility] Provide icon info to screen readers

### ISSUE THREAD

**Issue by jooyoungseo** [microsoft/vscode#163506]
CC @isidorn

As far as I know, sighted people can visually identify signature type (e.g., variable, class, function, method, etc.) in Document Symbol view (Ctrl+Shift+.).

Say, for example,

`test.py`

``` python
s = "Hello world!"

class Test():
    def foo():
        pass

def bar():
    pass
```

If you open the Document Symbol view (Ctrl+Shift+.) with screen reader (e.g., NVDA or JAWS)
 and navigate symbols with arrows, you won't hear any signature type. This means blind people do not know  whether currently focused
symbol is class or function or method, etc.

It would be great if this kind of information could be provided via either text or unique audio sound.

**Comment by isidorn** [comment]
Great feedback for the breadcrumbs view, thanks! Assigning to @jrieken 
Joh can we add the symbol type at the end of the breadcrumb element aria label?

fyi @meganrogge

**Comment by VSCodeTriageBot** [comment]
This feature request is now a candidate for our backlog. The community has 60 days to [upvote](https://github.com/microsoft/vscode/wiki/Issues-Triaging#up-voting-a-feature-request) the issue. If it receives 20 upvotes we will move it to our backlog. If not, we will close it. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/vscode-issue-lifecycle).

Happy Coding!

**Comment by jooyoungseo** [comment]
My recent pilot studies with other blind programmers have revealed the needs of this feature. I asked five blind programmers to identify the code hierarchy, including super class, sub class, and their member variables and methods. Those who used VSCode breadcrumbs view as a way of code glancing complained about the missing information on symbol types. I hope that this feature request can be seriously taken into consideration. this will dramatically improve the accessible "glanceability" for blind programmers.

**Comment by jooyoungseo** [comment]
@meganrogge

This improvement will become another game changer.

I have found that symbols have their unique class names like below:

* .codicon-symbol-class

* .codicon-symbol-constructor

* .codicon-symbol-property

* .codicon-symbol-method

* .codicon-symbol-variable

* .codicon-symbol-function

Using this information, I suggest providing the following format for screen reader users:

> `{sylbol_type}: {symbol}`

## Example

> class: foo

> function: bar

* Note: I suggest `symbol_type` as a prefix and this aligns with Problem View model where users see the severity level first (e.g., Error: xx; warning: yy; info: zz).

**Comment by jooyoungseo** [comment]
@meganrogge -- Well, I have thought about this design more deeply, and have realized that symbol-type prefix might not be a good idea.

It may be noisy and verbose when users want to skim through symbol names. Especially in Markdown file, it might look like "string: heading title" and users may not want to hear the symbol-type first in this case.

After using R extension's variable view, I felt that symbol-type as a suffix might be a better approach.

## Example

> `foo (class)`

> `bar (variable)`

![image](https://user-images.githubusercontent.com/19754711/232777717-d9683fd2-8b67-4d9a-ad14-c9056c0c6efd.png)

**Comment by meganrogge** [comment]
Would using audio cues be better to reduce noise even more?

**Comment by jooyoungseo** [comment]
@meganrogge -- I think mapping multiple symbol types with distinct audio cues might introduce cognitive burden until users get used to them. The ideal case would be:

1. Provide symbol types as a text suffix by default (1st milestone).

1. Add user setting to replace the text info with audio cues (2nd milestone).

1. Allow users to change default audio cues to their own easy-to-understand custom audio files (separate milestone).

I heavily rely on audio cues personally, however, we need to think about alternative and flexible customization when implementing audio cues to not exclude deaf-blind and hard-of-hearing population. We may want to avoid audio cue-only approach. I will keep using and suggesting audio cues. My point is the level of flexibility and customization for better accessibility.

**Comment by meganrogge** [comment]
@jooyoungseo said he interviewed blind devs and they said this was a big issue. 

@jrieken do you want help with this one or can we prioritize it in the next few iterations?

**Comment by meganrogge** [comment]
We use symbols all over the workbench - in the quick fix menus, in the `Go to symbol` quick pick, and more. We should fix this for all of those

**Comment by meganrogge** [comment]
@rperez030 I am thinking to work on this in July since it seems like a high impact item. I don't want to step on your toes though - lmk what you think

**Comment by rperez030** [comment]
Oh please feel free to take it. I'll be happy to review / test / lern from your fix. I can take on something else next week. I'm sure there is plenty for me to take on :).

**Comment by meganrogge** [comment]
To verify: 

- using a screen reader, use `Ctrl/Cmd+Shift+.` to focus the breadcrumbs view. 
- verify you are provided with the symbol info

**Comment by rperez030** [comment]
Verified. I'm loving it by the way. It makes navigating code way less stressing, so much that I can now fully understand how bad it was before. It is surprising how much little things can make a big difference. Thanks @jooyoungseo  for opening the issue in the first place and @meganrogge  for fixing it.

**Comment by jooyoungseo** [comment]
Yes, I concur. This feature is a game changer. Big thanks to @meganrogge for this improvement! @rperez030 -- Thanks for your nice verification.
